lib/colorls/core.rb

```diff
@@ -124,12 +150,64 @@ def display_report
         .colorize(:white)
     end
 
+    def mode_info(stat)
+      mode = ''
+      stat.mode.to_s(2).rjust(16, '0')[-9..-1].each_char.with_index do |c, i|
+        if c == '0'
+          mode += '-'.colorize(:gray)
+        else
+          case (i % 3)
+          when 0 then mode += 'r'.colorize(:yellow)
+          when 1 then mode += 'w'.colorize(:magenta)
+          when 2 then mode += 'x'.colorize(:cyan)
+          end
+        end
+      end
+      mode
+    end
+
+    def user_info(stat)
+      begin
+        user = Etc.getpwuid(stat.uid).name
+      rescue ArgumentError
+        user = stat.uid
+      end
+      user = user.to_s.ljust(@userlength, ' ')
+      user.colorize(:green) if user == Etc.getlogin
+    end
+
+    def group_info(stat)
+      begin
+        group = Etc.getgrgid(stat.gid).name
+      rescue ArgumentError
+        group = stat.gid
+      end
+      group.to_s.ljust(@grouplength, ' ')
+    end
+
+    def size_info(stat)
+      size = Filesize.from("#{stat.size} B").pretty.split(' ')
+      "#{size[0][0..-4].rjust(3,' ')} #{size[1].ljust(3,' ')}"
+    end
+
+    def mtime_info(stat)
+      mtime = stat.mtime.asctime
+      mtime = mtime.colorize(:yellow) if Time.now - stat.mtime < 24 * 60 * 60
+      mtime = mtime.colorize(:green)  if Time.now - stat.mtime < 60 * 60
+      mtime
+    end
+
+    def long_info(content)
+      stat = File.stat(content)
+      "#{mode_info(stat)} #{user_info(stat)} #{group_info(stat)} #{size_info(stat)} #{mtime_info(stat)}"
+    end
+
```

Comment:
Maybe using a "\t" instead of space between each variable would do a better job of indenting? Like, 

```ruby
"#{mode_info(stat)}\t#{user_info(stat)}\t#{group_info(stat)}\t#{size_info(stat)}\t#{mtime_info(stat)}"
```